src/app/filelogger.cpp

```diff
@@ -176,7 +176,7 @@ void FileLogger::flushLog()
 void FileLogger::openLogFile()
 {
     if (!m_logFile.open(QIODevice::WriteOnly | QIODevice::Append | QIODevice::Text)
-        || !m_logFile.setPermissions(QFile::ReadOwner | QFile::WriteOwner))
+        || (!(m_logFile.permissions() & (QFile::ReadOwner | QFile::WriteOwner)) && !m_logFile.setPermissions(QFile::ReadOwner | QFile::WriteOwner)))
```

Comment:
```suggestion
        || (!m_logFile.permissions().testFlags(QFile::ReadOwner | QFile::WriteOwner) && !m_logFile.setPermissions(QFile::ReadOwner | QFile::WriteOwner)))
```

Reply:
Is this expression correct?
If `m_logFile.permissions().testFlags` returns `true`, it will still fall into the `if` body.

Reply:
The new code also changes the [original intention](https://github.com/qbittorrent/qBittorrent/pull/22800#discussion_r2117506591), is it acceptable?

Reply:
> I suppose this alternative would be possible

👍

Reply:
I switched to the version that handles permission modification as best effort.